src/components/ui/weather-card/use-weather-card.ts

```diff
@@ -27,7 +29,15 @@ const useWeatherCard = ({ city }: WeatherCardProps) => {
     });
   }, []);
 
-  return { dateBuilder, data, isLoading };
+  const handleDeleteCard = () => {
+    setCities(
+      cities.filter(function (value, index, array) {
+        return index === 0;
+      })
+    );
+  };
```

Comment:
Your handle delete function doesn't work. Your filter function removes all cities from the array except the first city. Please update the logic.

Reply:
fixed